src/WebJobs.Script/Host/BlobLeaseManager.cs

```diff
@@ -13,26 +13,28 @@ namespace Microsoft.Azure.WebJobs.Script
 {
     internal sealed class BlobLeaseManager : IDisposable
     {
-        private const string LockBlobName = "hostlock";
-        private const string HostContainerName = "azure-functions-host";
+        internal const string LockBlobName = "hostlock";
+        internal const string HostContainerName = "azure-functions-host";
 
         private readonly Timer _timer;
         private readonly TimeSpan _leaseTimeout;
         private readonly TimeSpan _renewalInterval;
         private readonly TimeSpan _leaseRetryInterval;
         private readonly TraceWriter _traceWriter;
         private readonly string _hostId;
+        private readonly string _instanceId;
         private ICloudBlob _lockBlob;
         private string _leaseId;
         private bool _disposed;
         private bool _processingLease;
 
-        internal BlobLeaseManager(ICloudBlob lockBlob, TimeSpan leaseTimeout, string hostId, TraceWriter traceWriter, TimeSpan? renewalInterval = null)
+        internal BlobLeaseManager(ICloudBlob lockBlob, TimeSpan leaseTimeout, string hostId, string instanceId, TraceWriter traceWriter, TimeSpan? renewalInterval = null)
         {
             _lockBlob = lockBlob;
             _leaseTimeout = leaseTimeout;
             _traceWriter = traceWriter;
             _hostId = hostId;
+            _instanceId = instanceId;
 
             // Renew the lease three seconds before it expires
```

Comment:
When you attempt a renewal, if it fails do you just give up and lose the lease? I'm thinking we'll want to do something more sophisticated in the future, similar to what SingletonManager does. It starts trying to renew once half the lease duration has expired, and uses a linear speedup strategy to retry progressively more often on failure.